linera-core/src/worker.rs

```diff
@@ -667,14 +695,32 @@ where
             .await?;
         // Persist certificate and hashed certificate values.
         for value in hashed_certificate_values {
-            self.cache_recent_value(Cow::Borrowed(value)).await;
+            self.cache_recent_hashed_certificate_value(Cow::Borrowed(value))
+                .await;
         }
-        let (result_hashed_certificate_value, result_certificate) = tokio::join!(
+
+        let pending_blobs = &chain.manager.get().pending_blobs;
```

Comment:
We need to make sure we only write the blobs that are actually referred to by this confirmed block certificate. The blobs that are pending because of other block proposals that didn't get finalized should not be stored.